Source/Core/Core/HW/EXI/BBA/BuiltIn.cpp

```diff
@@ -298,18 +302,30 @@ void CEXIETHERNET::BuiltInBBAInterface::HandleTCPFrame(const Common::TCPPacket&
   sf::IpAddress target;
   StackRef* ref = GetTCPSlot(tcp_header.source_port, tcp_header.destination_port,
                              Common::BitCast<u32>(ip_header.destination_addr));
-  const u16 properties = ntohs(tcp_header.properties);
-  if (properties & (TCP_FLAG_FIN | TCP_FLAG_RST))
+  const u16 flags = ntohs(tcp_header.properties) & 0xfff;
+  if (flags & (TCP_FLAG_FIN | TCP_FLAG_RST))
   {
     if (ref == nullptr)
       return;  // not found
 
-    ref->ack_num++;
+    ref->ack_num += 1 + static_cast<u32>(data.size());
```

Comment:
All the other `ack_num` values being set use `ntohl` did you intentionally leave it off here for some reason?

Reply:
`ack_num` is using the host's endianness and is used to acknowledge the TCP packet bytes (sequence number) received. When inspecting TCP packets from the BBA buffer we need to convert from _network to host_ (`ntoh*`) endianness. Here we're just incrementing it (not setting it from a network packet) so we don't need to swap it.